api/api/accounts/test_update_users.py

```diff
@@ -0,0 +1,307 @@
+from collections import OrderedDict
+
+import pytest
+from django.contrib.auth import get_user_model
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from .models import Faculty, Staff, Student
+
+User = get_user_model()
+primary_test_user_mail = 'test@user.com'
+
+
+@pytest.mark.django_db
+@pytest.fixture(scope='function')
+def test_user(django_user_model):
+    return django_user_model.objects.create(email=primary_test_user_mail,
+                                            first_name='test',
+                                            last_name='user')
+
+
+@pytest.fixture(scope='function')
+def client():
+    return APIClient()
+
+
+@pytest.mark.django_db
+def testUpdateUserView_nonOwnerRequest_throwsForbidden(client, test_user):
+
+    # given
+    client.force_authenticate(test_user)
+    new_user = User.objects.create(email='new@user.com',
+                                   first_name='new',
+                                   last_name='user')
+
+    # when
+    response = client.put(f'/users/{new_user.id}/', {'email': 'changed_mail@user.com'})
+
+    # then
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+@pytest.mark.django_db
+def testUpdateUserView_baseUser_ownerChangeRequest_updatesSuccessful(client, test_user):
+
+    # given
+    client.force_authenticate(test_user)
+
+    # when
+    response = client.put(f'/users/{test_user.id}/', {'email': 'user@user.com',
+                                                      'last_name': 'changed_name'})
+
+    user = User.objects.get(id=test_user.id)
+    # then
+    assert response.status_code == status.HTTP_200_OK
+    assert response.data['last_name'] == 'changed_name'
+    assert user.last_name == 'changed_name'
+
+
+@pytest.mark.django_db
+def testUpdateUserView_baseUser_adminChangeRequest_updatesSuccessful(client, test_user):
+
+    # given
+    admin_user = User.objects.create(email='admin@user.com',
+                                     first_name='admin',
+                                     is_staff=True)
+
+    client.force_authenticate(admin_user)
+
+    # when
+    response = client.put(f'/users/{test_user.id}/', {'email': 'test@user.com',
+                                                      'first_name': 'changed_name'})
+    test_user = User.objects.get(id=test_user.id)
```

Comment:
I believe this should move to `# then` section as it's a result of calling response

Should be called `updated_test_user`